core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      const { getByTestId, unmount } = render(<Input name="test" type="number" />);
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      expect(wrapper.style.minWidth).toBe('');
+      unmount();
+
+      const { getByTestId: getByTestId2 } = render(<Input name="test" type="text" />);
+      const wrapper2 = getByTestId2('DesignSystem-InputWrapper');
+      expect(wrapper2.style.minWidth).toBe('256px');
+    });
+
+    it('sets custom minWidth when provided', () => {
+      const { getByTestId } = render(<Input name="test" minWidth="300px" />);
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      expect(wrapper.style.minWidth).toBe('300px');
+    });
+  });
+
+  describe('Event handlers', () => {
+    it('calls onChange handler when input value changes', () => {
+      const onChangeMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" onChange={onChangeMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.change(input, { target: { value: 'new value' } });
+
+      expect(onChangeMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls onFocus handler when input gains focus', () => {
+      const onFocusMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" onFocus={onFocusMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.focus(input);
+
+      expect(onFocusMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls onBlur handler when input loses focus', () => {
+      const onBlurMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" onBlur={onBlurMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.blur(input);
+
+      expect(onBlurMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls onClick handler when input is clicked', () => {
+      const onClickMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" onClick={onClickMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.click(input);
+
+      expect(onClickMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls onPaste handler when content is pasted', () => {
+      const onPasteMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" onPaste={onPasteMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.paste(input, { clipboardData: { getData: () => 'pasted text' } });
+
+      expect(onPasteMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('focuses input when wrapper is clicked', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+
+      input.focus = jest.fn();
+
+      fireEvent.click(wrapper);
+      expect(input.focus).toHaveBeenCalled();
+    });
+  });
+
+  describe('Clear functionality', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.